tests/substrate.rs

```diff
@@ -188,11 +191,12 @@ impl Externals for MockSubstrate {
                 Ok(None)
             }
             Some(SubstrateExternal::seal_get_storage) => {
-                assert_eq!(args.len(), 3);
+                assert_eq!(args.len(), 4);
 
                 let key_ptr: u32 = args.nth_checked(0)?;
-                let dest_ptr: u32 = args.nth_checked(1)?;
-                let len_ptr: u32 = args.nth_checked(2)?;
+                //let key_len: u32 = args.nth_checked(1)?; TODO
```

Comment:
The MockSubstrate assumes that the key is 32 bytes. I think this should either do `assert_eq!(key_len, 32);` or implement variable-length key lengths.

Reply:
If it assumes 32bit key length for now I can check this. But I'd rather implement all the variable Key length stuff in another PR.

Reply:
Where is the key length check? The code below assumes the key is 32 bytes, see `let mut key: StorageKey = [0; 32];`